scripts/reference_disks/create_images.sh

```diff
@@ -0,0 +1,390 @@
+
+Dependencies:
+* \`sudo\`
+* \`gcloud\`
+* \`jq\`
+USAGE
+    exit 1
+}
+
+# Date part to add to image names
+DATE_STR=$(date +"%Y-%m-%d")
+
+# The block size to divide images, aka all images will be a multiple of 5 (metric) gigabytes
+BLOCK_GBS=5
+BLOCK_SIZE=$((BLOCK_GBS * 10 ** 9))
+
+DRY_RUN=true
+EXISTING_RESOURCES=false
+MISSING_DEPENDENCIES=false
+
+error_msg() {
+    echo "$(tput setaf 1)ERROR:$(tput sgr0)" "$@" >&2
+}
+
+# Convert an image name prefix to an image name
+get_image_name() {
+    local image_name_prefix
+
+    image_name_prefix="$1"
+    # Not 100% sure of the origin of this naming scheme. For now, the naming scheme is copied from
+    # https://github.com/broadinstitute/firecloud-develop/blob/6705a3d7cccc9f536dff78a8359b6d4b586e32ab/base-configs/cromwell/cromwell-reference-images.conf#L3
+    echo "$image_name_prefix-public-$DATE_STR"
+}
+
+# Check for already existing resources
+check_existing() {
+    local image_name
+
+    image_name="$1"
+
+    if [[ -n "$(
+        gcloud compute images list --filter "name=('$image_name')" --format "value(name)"
+    )" ]]; then
+        error_msg "Existing image. Use \`gcloud compute images delete $image_name\`"
+        EXISTING_RESOURCES=true
+    fi
+
+    if mount | grep "/mnt/disks/$image_name" &>/dev/null; then
+        error_msg "Mounted disk. Use \`sudo umount /mnt/disks/$image_name\`"
+        EXISTING_RESOURCES=true
+    fi
+
+    if [[ -n "$(
+        gcloud compute instances list --filter "disks.deviceName=('$image_name')" --format="value(disks[].deviceName)"
+    )" ]]; then
+        error_msg "Attached disk. Use \`gcloud compute instances detach-disk $(hostname) --disk $image_name\`"
+        EXISTING_RESOURCES=true
+    fi
+
+    if [[ -n "$(
+        gcloud compute disks list --filter "name=('$image_name')" --format "value(name)"
+    )" ]]; then
+        error_msg "Existing disk. Use \`gcloud compute disks delete $image_name\`"
+        EXISTING_RESOURCES=true
+    fi
+
+    if [[ -e "$image_name.manifest.conf" ]]; then
+        error_msg "Existing manifest. Use \`rm '$image_name.manifest.conf'\`"
+        EXISTING_RESOURCES=true
+    fi
+}
+
+# Convert the retrieved base64 value to decimal since that's what CromwellRefdiskManifestCreatorApp does, slowly
+# https://github.com/broadinstitute/cromwell/blob/65/CromwellRefdiskManifestCreator/src/main/java/org/broadinstitute/manifestcreator/CromwellRefdiskManifestCreatorApp.java#L116
+get_cloud_hash() {
+    local cloud_object
+
+    cloud_object="$1"
+
+    # URL encode the object using jq. Easier than trying to figure out which python2/python3 is available.
+    # And we're using jq again just below to retrieve the crc32c from the output.
+    # https://stackoverflow.com/questions/37309551/how-to-urlencode-data-into-a-url-with-bash-or-curl#comments-37309630
+    cloud_object_encoded="$(jq --raw-input --raw-output @uri <<<"$cloud_object")"
+
+    # No need to waste IO & CPU. The crc32c is already in the cloud
```

Comment:
TOL not sure if the original author thought an independent check of the CRC32c was necessary for a `gsutil cp`'d file, but that's certainly not the assumption we've been making recently in DRS-land...

Reply:
No clue about the original intent. The hashes this script generated for the hg19 files match the decimals in fc-dev, and shaved 2h40 minutes off the run on my GCE instance.